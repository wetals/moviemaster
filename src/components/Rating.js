import React from "react";

const Rating = props => {
  const { rating } = props;

  const updateRating = event => {
    props.updateRating(event.target.value);
  };

  return (
    <div className="rating">
      <h3 className="title">Rating</h3>
      <div className="rating-slider">
        <input
          className="rating-slider__range"
          type="range"
          value={rating}
          min="0"
          max="10"
          step="0.5"
          onChange={updateRating}
        />
        <span className="rating-slider__value">{rating}</span>
      </div>
    </div>
  );
};

export default Rating;
