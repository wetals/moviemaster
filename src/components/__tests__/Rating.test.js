import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import Rating from "../Rating";

test("Rating renders a snapshot properly", () => {
  const tree = renderer.create(<Rating />).toJSON();
  expect(tree).toMatchSnapshot();
});
