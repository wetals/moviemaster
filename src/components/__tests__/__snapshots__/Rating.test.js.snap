// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating renders a snapshot properly 1`] = `
<div
  className="rating"
>
  <h3
    className="title"
  >
    Rating
  </h3>
  <div
    className="rating-slider"
  >
    <input
      className="rating-slider__range"
      max="10"
      min="0"
      onChange={[Function]}
      step="0.5"
      type="range"
      value={undefined}
    />
    <span
      className="rating-slider__value"
    />
  </div>
</div>
`;
